dbt/include/synapse/macros/adapters.sql

```diff
@@ -0,0 +1,142 @@
+{% macro synapse__information_schema_name(database) -%}
+  {{ return(sqlserver__information_schema_name(database)) }}
+{%- endmacro %}
+
+{% macro synapse__get_columns_in_query(select_sql) %}
+    {% call statement('get_columns_in_query', fetch_result=True, auto_begin=False) -%}
+        select TOP 0 * from (
+            {{ select_sql }}
+        ) as __dbt_sbq
+    {% endcall %}
+    {{ return(load_result('get_columns_in_query').table.columns | map(attribute='name') | list) }}
+{% endmacro %}
+
+{% macro synapse__list_relations_without_caching(schema_relation) %}
+  {{ return(sqlserver__list_relations_without_caching(schema_relation)) }}
+{% endmacro %}
+ 
+{% macro synapse__list_schemas(database) %}
+  {{ return(sqlserver__list_schemas(database)) }}
+{% endmacro %}
+
+{% macro synapse__create_schema(relation) -%}
+  {% call statement('create_schema') -%}
+    IF NOT EXISTS (SELECT * FROM sys.schemas WHERE name = '{{ relation.without_identifier().schema }}')
+    BEGIN
+    EXEC('CREATE SCHEMA {{ relation.without_identifier().schema }}')
+    END
+  {% endcall %}
+{% endmacro %}
+
+{% macro synapse__drop_schema(relation) -%}
+  {{ return(sqlserver__drop_schema(relation)) }}
+{% endmacro %}
+
+{# TODO make this function just a wrapper of synapse__drop_relation_script #}
+{% macro synapse__drop_relation(relation) -%}
+  {{ return(sqlserver__drop_relation(relation)) }}
+{% endmacro %}
```

Comment:
CI is failing with this message right now and I'm not sure why....
```
=================================== FAILURES ===================================
___________________________ usecase: test_dbt_empty ____________________________
Unknown error: Compilation Error in macro statement (macros/core.sql)
  'relation' is undefined
  
  > in macro sqlserver__drop_schema (macros/adapters.sql)
  > called by macro drop_schema (macros/adapters/common.sql)
  > called by macro statement (macros/core.sql)
```

Reply:
Hm... I'm not sure what's going on here, and whether it's an issue with the inheritance/argument passage or with the macro itself. What happens if you reimplement (copy-paste) to code for [`sqlserver__drop_schema`](https://github.com/dbt-msft/dbt-sqlserver/blob/53134ce95e6d6e6924bba960194495f54ed3b3f5/dbt/include/sqlserver/macros/adapters.sql#L54-L72) as `synapse__drop_schema`?

Reply:
so I got the same result. I started playing around with it and got the helpful error message below. `sp_rename` is used in `sqlserver__rename_relation` but not in `synapse__rename_relation`. so it seems like an inheritance issue?
```
An insufficient number of arguments were supplied for the procedure or function sp_rename. 
```
 I tried to uninstall and re-install dbt-synapse with `pip install -e .` but not I'm getting the 2nd error below...
```
Running with dbt=0.18.1
Encountered an error:
module 'dbt.adapters.sqlserver' has no attribute 'Plugin'
```

Reply:
Ha, well, both of those are confusing to me!

Reply:
I'm gonna sleep on it... thanks for being so patient and helpful through all this. So close now -- I can taste the light at the end of the tunnel